web_src/js/markup/anchors.js

```diff
@@ -30,6 +30,24 @@ export function initMarkupAnchors() {
     });
     heading.prepend(a);
   }
+  for (const anchor of document.querySelectorAll('.markup a[href^="#"]')) {
+    const href = anchor.getAttribute('href');
+    if (!href.startsWith('#user-content-')) {
+      continue;
+    }
+    const originalId = href.replace(/^#user-content-/, '');
+    anchor.setAttribute('href', `#${encodeURIComponent(originalId)}`);
+    const anchorDest = document.getElementsByName(originalId);
+    if (anchorDest?.length === 1) {
+      anchor.addEventListener('click', () => {
+        anchorDest[0].scrollIntoView();
+      });
+    } else {
+      anchor.addEventListener('click', (e) => {
+        scrollToAnchor(e.currentTarget.getAttribute('href'), false);
+      });
+    }
```

Comment:
```suggestion
```

I see no difference when removing this block because once the `href` on the source link is corrected, it will work natively. Would you agree removing this part?